curriculum/js3/2.js

```diff
@@ -4,8 +4,60 @@
  * @param return {number}
  */
 
+/* ver 1 - forEach() & reduce()
 const solution = (arr)=>{
-  return 0;
+  if (!arr.length) return null;
+
+  let obj = {};  //key -> number, value -> frequency
+  arr.forEach(el => {
+    obj[el] = (obj[el] === undefined) ? 1 : obj[el]+1;
+  });
+  
+  let values = Object.values(obj);
+  let max = 0;
+
+  //get index of highest value
+  let i = values.reduce((acc, el, idx) => {
+    if (el > max) {
+      max = el;
+      return idx;
+    }
+    else {
+      return acc;
+    }
+  },0);
+
+  //key is the number
+  return parseInt(Object.keys(obj)[i]);
+}
+*/
+
+/* ver 2 - recursive */
+const buildFreq = (a,obj={}) => {
+  if (!a.length) return obj;
+  obj[a[0]] = obj[a[0]] === undefined ? 1 : obj[a[0]]+1;
+  return buildFreq(a.slice(1),obj);
```

Comment:
try not to use slice (which modifies an array), its slower. Use i=0 as an index so you never change array.